### PR TITLE
Fix account inbox implementation

### PR DESCRIPTION
**Pull request by turbolent**
## Description

- Public accounts do not have an inbox, remove related code
- Add missing `inbox` field name to AuthAccount type
- Fix static type of AuthAccount.Inbox
- Add missing mappings between sema and primitive static types
- Add tests to avoid similar omissions for future primitive static types

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/interpreter/account.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ var authAccountFieldNames = []string{
 	sema.AuthAccountAddressField,
 	sema.AuthAccountContractsField,
 	sema.AuthAccountKeysField,
+	sema.AuthAccountInboxField,
```
Comment (dsainati1) [inline runtime/interpreter/account.go:36]: Confused by this; what effect does this actually have given that none of the interpreter tests failed without this? I would have expected not to be able to access the `inbox` field without specifying this.

File: runtime/interpreter/accountinbox.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ import (
 // AuthAccountInbox
 
 var authAccountInboxTypeID = sema.AuthAccountInboxType.ID()
-var authAccountInboxStaticType StaticType = PrimitiveStaticTypeAuthAccountKeys
+var authAccountInboxStaticType StaticType = PrimitiveStaticTypeAuthAccountInbox
```
Comment (dsainati1) [inline runtime/interpreter/accountinbox.go:31]: Oops! Thanks for catching this

**Review by dsainati1 — APPROVED** [review]
Sorry about this; guess I forgot to remove some of the public account stuff after we changed the proposal in the FLIP again. There are definitely benefits to implementing the FLIPs as they are being designed as a prototype of sorts, but the downside is it gets a lot harder to keep track of what actually belongs in the implementation PRs 😅 .

**Comment by turbolent** [comment]
No worries! I had missed these things in the review too and it's easy to fix 🙂 
We probably need more checks to prevent such accidents, not quite sure yet how though

File: runtime/interpreter/account.go (hunk 1 above)
Comment (turbolent) [reply runtime/interpreter/account.go:36]: See the usage of `FieldNames` in `SimpleCompositeValue`. For example, they are used to iterate over the fields, prepare a string representation, check static type conformance, etc

File: runtime/convertValues.go [hunk 3]
```diff
@@ -529,9 +529,7 @@ func exportSimpleCompositeValue(
 		structure, err := cadence.NewMeteredStruct(
 			inter,
 			len(fieldNames),
-			func() ([]cadence.Value, error) {
-				return makeFields()
-			},
+			makeFields,
```
Comment (dreamsmasher) [inline runtime/convertValues.go:532]: nice eta reduce

File: runtime/interpreter/accountinbox.go (hunk 2 above)
Comment (dreamsmasher) [reply runtime/interpreter/accountinbox.go:31]: AAAA that's where my failing test was coming from, I knew it stemmed from static type resolution *somewhere*

**Review by dreamsmasher — APPROVED** [review]

**Pull request merged**